test/extensions/tradeExtension.spec.ts

```diff
@@ -0,0 +1,233 @@
+import "module-alias/register";
+
+import { Address, Account } from "@utils/types";
+import { ADDRESS_ZERO } from "@utils/constants";
+import { DelegatedManager, TradeExtension } from "@utils/contracts/index";
+import { SetToken, TradeModule } from "@setprotocol/set-protocol-v2/utils/contracts";
+import DeployHelper from "@utils/deploys";
+import {
+  addSnapshotBeforeRestoreAfterEach,
+  ether,
+  getAccounts,
+  getWaffleExpect
+} from "@utils/index";
+import { SystemFixture } from "@setprotocol/set-protocol-v2/utils/fixtures";
+import { getSystemFixture, getRandomAccount } from "@setprotocol/set-protocol-v2/utils/test";
+import { ContractTransaction } from "ethers";
+
+const expect = getWaffleExpect();
+
+describe("TradeExtension", () => {
+  let owner: Account;
+  let methodologist: Account;
+  let operator: Account;
+  let factory: Account;
+
+  let deployer: DeployHelper;
+  let setToken: SetToken;
+  let setV2Setup: SystemFixture;
+
+  let tradeModule: TradeModule;
+
+  let delegatedManager: DelegatedManager;
+  let tradeExtension: TradeExtension;
+
+  before(async () => {
+    [
+      owner,
+      methodologist,
+      operator,
+      factory
+    ] = await getAccounts();
+
+    deployer = new DeployHelper(owner.wallet);
+
+    setV2Setup = getSystemFixture(owner.address);
+    await setV2Setup.initialize();
+
+    tradeModule = await deployer.setDeployer.modules.deployTradeModule(setV2Setup.controller.address);
```

Comment:
do we want to make the `SystemFixture` more comprehensive? To include things like the `TradeModule` (or maybe all modules).